### PR TITLE
Add Teleports methods

### PR DESCRIPTION
**Pull request by Minazuki3230**

### CODE REVIEW

**Comment by louis1706** [comment]
i don't thinks it's really needed in Exiled

File: Exiled.API/Features/Player.cs [hunk 1]
```diff
@@ -2212,6 +2212,30 @@ public void Reconnect(ushort newPort = 0, float delay = 5, bool reconnect = true
...
+        /// <summary>
+        /// Teleports player to <see cref="RandomObjectType"/>.
+        /// </summary>
+        /// <param name="type">Object for teleport.</param>
+        public void RandomTeleport(RandomObjectType type)
+        {
+            if (type == RandomObjectType.Door)
+                Teleport(Door.List.ElementAt(UnityEngine.Random.Range(0, Door.List.Count())));
+            else if (type == RandomObjectType.Room)
+                Teleport(Door.List.ElementAt(UnityEngine.Random.Range(0, Door.List.Count())));
```
Comment (BuildBoy12) [inline Exiled.API/Features/Player.cs]: Door.List usage in Room request

**Comment by joker-119** [comment]
> i don't thinks it's really needed in Exiled

I agree but I removed the things causing build errors and made these actually useful, and expandable in the future.

**Review by BuildBoy12 — CHANGES_REQUESTED** [review]

File: Exiled.API/Features/Player.cs (hunk 1 above)
Comment (joker-119) [reply Exiled.API/Features/Player.cs]: re-read the PR, I changed everything in my commit.

**Comment by louis1706** [comment]
It's going to teleport player in the killerzone of Tesla gate

File: Exiled.API/Features/Player.cs [hunk 2]
```diff
@@ -2212,6 +2216,91 @@ public void Reconnect(ushort newPort = 0, float delay = 5, bool reconnect = true
         /// <inheritdoc cref="Map.GetNearCameras(Vector3, float)"/>
         public IEnumerable<Camera> GetNearCameras(float toleration = 15f) => Map.GetNearCameras(Position, toleration);
 
+        /// <summary>
+        /// Teleports the player to the given <see cref="Vector3"/> coordinates.
+        /// </summary>
+        /// <param name="position">The <see cref="Vector3"/> coordinates to move the player to.</param>
+        public void Teleport(Vector3 position) => Position = position;
```
Comment (BuildBoy12) [inline Exiled.API/Features/Player.cs:2223]: Not sure I see the use in having another method to access set_Position

**Comment by joker-119** [comment]
> It's going to teleport player in the killerzone of Tesla gate

Not my problem :smirk: Idk if tesla.transform.Forward is towards the sides of the gate or actually infront of it

File: Exiled.API/Features/Player.cs (hunk 2 above)
Comment (joker-119) [reply Exiled.API/Features/Player.cs:2223]: I don't either but if we're ganna add teleport methods we might aswell have this? 
I think
Teleport(door.Position + vector3.up)
looks better than
Position = door.Position + Vector3.up
but that's just me, personally. We could remove this teleport method.

**Comment by louis1706** [comment]
> > It's going to teleport player in the killerzone of Tesla gate
> 
> Not my problem :smirk: Idk if tesla.transform.Forward is towards the sides of the gate or actually infront of it

You can see it with the rotation of the TeslGate

**Comment by joker-119** [comment]
> > > It's going to teleport player in the killerzone of Tesla gate
> > 
> > 
> > Not my problem smirk Idk if tesla.transform.Forward is towards the sides of the gate or actually infront of it
> 
> You can see it with the rotation of the TeslGate

But idk off-hand what to do to calculate the safe area infront of the tesla, which is why it just sends them to the center for now, we will need to make adjustments for each room eventually based on rotation, so we can do gates then to. This is just proof of concept because the original PR was a good idea executed very poorly

**Comment by Minazuki3230** [comment]
Now the player will not be teleported to the killerzone of TeslaGate.

**Comment by joker-119** [comment]
What the actual fuck?

**Comment by louis1706** [comment]
The random you have put going alway return you 1 and never 2 and never 0

**Comment by joker-119** [comment]
I fixed it. Again.

**Comment by Minazuki3230** [comment]
Bruh
image

**Comment by Minazuki3230** [comment]
Bruh
gif

**Comment by Nekonyx** [comment]
Bruh
I would like to see overloads instead of mysterious `object`

**Comment by joker-119** [comment]
> Bruh gif P.S. In the test plugin I registered a teleport on Player.CancellingItemUse

I'll adjust the teleport point more when I have time to get ingame and test it myself.

**Comment by Nekonyx** [comment]
> > Bruh I would like to see overloads instead of mysterious `object`
> 
> Doing overloads or an enum for it is adding unnecessary clutter, this way we can add various object types and the syntax is very simple to understand, player.Teleport(literally any object here) It's not mysterious, it's polymorphism, a basic concept of any OO language.

Then there should be at least details in description.

**Comment by Minazuki3230** [comment]
> > Bruh gif P.S. In the test plugin I registered a teleport on Player.CancellingItemUse
> 
> I'll adjust the teleport point more when I have time to get ingame and test it myself.

https://github.com/Exiled-Team/EXILED/pull/1125/commits/e9f94ec7d3d81ab45383d1fcbb1cc9ac82f13987 worked in 8 out of 8 rounds.

**Comment by joker-119** [comment]
> > > Bruh gif P.S. In the test plugin I registered a teleport on Player.CancellingItemUse
> > 
> > 
> > I'll adjust the teleport point more when I have time to get ingame and test it myself.
> 
> [e9f94ec](https://github.com/Exiled-Team/EXILED/commit/e9f94ec7d3d81ab45383d1fcbb1cc9ac82f13987) worked in 8 out of 8 rounds.

Yes but it was a stupid way of doing it.

**Pull request merged**